src/lib/ActionClient.js

```diff
@@ -30,6 +31,10 @@ class ActionClient extends EventEmitter {
       GoalID = msgUtils.requireMsgPackage('actionlib_msgs').msg.GoalID;
     }
 
+    if (GoalID === null) {
```

Comment:
@mjyc I agree with @chris-smith here. Looks like you're intending to check and then set the `Header`. I'd be happy to merge after the change.